implementations/rust/channel/src/lib.rs

```diff
@@ -23,54 +23,405 @@
 extern crate ockam_common;
 
 use error::*;
-use ockam_kex::{error::KeyExchangeFailErrorKind, CompletedKeyExchange, KeyExchanger};
-use std::io::{Read, Write};
-
-/// Represents an Ockam channel for reading and writing payloads
-#[derive(Debug)]
-pub struct Channel<R: Read, W: Write, KX: KeyExchanger> {
-    exchange_data: Option<CompletedKeyExchange>,
-    key_exchanger: KX,
-    reader: R,
-    writer: W,
+use failure::_core::marker::PhantomData;
```

Comment:
`
```suggestion
use core::marker::PhantomData;
```
I will update and push